src/nvim/api/vim.c

```diff
@@ -790,7 +793,11 @@ void nvim_echo(Array chunks, Boolean history, Dict(echo_opts) *opts, Error *err)
     verbose_enter();
   }
 
-  msg_multihl(hl_msg, history ? "echomsg" : "echo", history);
+  if (opts->err) {
+    emsg_multiline(chunks.items[0].data.array.items[0].data.string.data, true);
```

Comment:
Should we error when `chunks.size > 1` or `history == false` or just silently execute the documented behavior?

Reply:
Why not print all chunks?

Reply:
`nvim_echo` supports multiple chunks to give a different highlight to different chunks. For `opts->err` I think it makes sense to implicitly highlight with `ErrorMsg`. I suppose we could still support multiple chunks though, even if that doesn't serve any purpose for the caller to pass in multiple chunks?

Reply:
> If people want to send chunks with random highlights, we might as well let them

With the connotation that those highlight's aren't actually applied? Wouldn't it be better to error in that case?

Or are you saying that `opts->err` with different highlight chunks _should_ apply those highlights? In that case I suppose we would need to add a `hl_id` argument to `emsg_multiline`.

Reply:
> In that case I suppose we would need to add a `hl_id` argument to `emsg_multiline`.

Oh, let's not bother with that. For now we'll just document that chunk highlights are ignored. Or we could throw an error.

But at least supporting multiple chunks seems harmless?

Reply:
Harmless yes, but also pointless. Indicating to me that an error might be more appropriate. Though supporting different highlight chunks even for `opts->err` does sound the least surprising/easiest to document to me. I don't think the change would be too involved.

Reply:
> I don't think the change required would be too involved.

SGTM